### ISSUE TITLE
4. Deleted goals list

### ISSUE THREAD

**Issue by Dartv** [Pliohub/Plio#731]
![](https://i.gyazo.com/a273376b699d898cd9b5788e28aa76bb.png)
They should be displayed underneath timeline. Refer to http://ui2.isometrica.io/ui1-3/ for example.

**Comment by Praxie** [comment]
@sives 

1. Is this required?
2. Do we need to store all deleted goals there?
2.1. There is no way to set the minimal set of goals displayed.
2.2. Should there be a "View all items (X more)" button?
3. Do we need to display a modal with read-only removed goal information on click? (Note there there are no read-only version of the Goal card because we have no Listview, so we'll need to create one). We can display the regular Edit Goal card or nothing at all.

**Comment by Dartv** [comment]
2.3. There is no way to completely remove goal or milestone (they're gonna stack for years)

**Comment by sives** [comment]
Yes, this is required.   I agree that this list could get very long and unmanageable, so I've made some design proposals below.

Regarding your question 2.1, there should be a way to change this number from the default number in the Key goals sub card in Org settings (see below).   The default number should be 5.

I suggest we change the title of the Deleted goals list to "Completed & deleted goals".  Regarding your question 2.2, there should be a "View all items (20 more)" button at the bottom of the list.  

A deleted goal will be indicated with the text "KG4: Launch insurance product (deleted 5 May 2016)
A completed goal will indicated as" KG4: Launch insurance product (completed 5 May 2016)  

There should be a Restore button and a Delete button at the right hand side of each item in this list.  Clicking on the Restore button returns the Key goal to the primary panel at the top, and changes its status to "Open - awaiting completion" or "Open - overdue".   Clicking on the Delete button gives the user a prompt asking them "Are you sure?" to confirm if they wish to delete this item forever (using similar text strings as used in deleting items from our Deleted item views in the existing system).  

I've updated Section 6 of the Plio Step 4 requirements document as follows:

There is a new subcard in Org settings called Key goals.  It allows the user to do two things.  Firstly, switch the scale for the Key goals display between the following 5 options in a standard dropdown menu (labelled "horizontal scale"):
- 1 month, 3 months (default), 6 months, 9 months, 12 months
Secondly to change the default number of Key goals displayed by default in the top panel.  Thirdly to change the number of Key goals displayed by default in the bottom panel (containing completed and deleted Key goals).
